File: java/src/main/java/com/languageweaver/sdk/samples/translations/file/batch/BatchFileTranslationService.java
package com.languageweaver.sdk.samples.translations.file.batch;

import com.languageweaver.sdk.common.LanguageWeaverClient;
import com.languageweaver.sdk.common.SdkFactory;
import com.languageweaver.sdk.common.configurations.ClientConfiguration;
import com.languageweaver.sdk.common.constants.PdfConverter;
import com.languageweaver.sdk.translate.common.request.TranslateBatchFileRequest;

import java.io.IOException;
import java.nio.file.Paths;
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.ExecutionException;

public class BatchFileTranslationService {

    public static void main(String[] args) throws Exception {
        try (LanguageWeaverClient lwClient = new SdkFactory().getLanguageWeaverClient(new ClientConfiguration())) {
            translateBatchFile(lwClient);
            translateBatchFileWithLinguisticOptions(lwClient);
        }
    }

    private static void translateBatchFile(LanguageWeaverClient lwClient) throws IOException, InterruptedException, ExecutionException {
        TranslateBatchFileRequest translateFileRequest = new TranslateBatchFileRequest()
                .setSourceLanguageId("eng")
                .setTargetLanguageId("fra")
                .setModel("generic")
                .setPdfConverter(PdfConverter.STANDARD)
                // provide list of dictionaries
                .addDictionary("dictionaryId or name")
                // provide full path to the input and output folders
                .setInputBatchFolderPath(Paths.get("java", "src", "main", "resources", "input").toFile().getAbsolutePath())
                .setOutputBatchFolderPath(Paths.get("java", "src", "main", "resources", "output").toFile().getAbsolutePath());

        lwClient.translateBatchFile(translateFileRequest);
    }

    private static void translateBatchFileWithLinguisticOptions(LanguageWeaverClient lwClient) throws IOException, InterruptedException, ExecutionException {
        Map<String, String> linguisticOptions = new HashMap<>();
        linguisticOptions.put("Spelling", "US");
        TranslateBatchFileRequest translateFileRequest = new TranslateBatchFileRequest()
                .setSourceLanguageId("fra")
                .setTargetLanguageId("eng")
                .setModel("generic")
                // provide list of dictionaries
                .addDictionary("dictionaryId or name")
                .setLinguisticOptions(linguisticOptions)
                // provide full path to the input and output folders
                .setInputBatchFolderPath(Paths.get("java", "src", "main", "resources", "input").toFile().getAbsolutePath())
                .setOutputBatchFolderPath(Paths.get("java", "src", "main", "resources", "output").toFile().getAbsolutePath());

        lwClient.translateBatchFile(translateFileRequest);
    }
}

File: dotnet/Samples/Translations/File/Batch/BatchFileTranslationService.cs
using LanguageWeaver.Sdk.Common;
using LanguageWeaver.Sdk.Configurations;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.Translate.Common.Request;

namespace LanguageWeaver.Sdk.Samples.Translations.File.Batch;

public class BatchFileTranslationService
{
    public static void Main()
    {
        using var lwClient = new SdkFactory().GetLanguageWeaverClient(new ClientConfiguration());

        TranslateBatchFile(lwClient);
        TranslateBatchFileWithLinguisticOptions(lwClient);
    }

    private static void TranslateBatchFile(ILanguageWeaverClient lwClient)
    {
        var translateBatchFileRequest = new TranslateBatchFileRequest
        {
            SourceLanguageId = "eng",
            TargetLanguageId = "fra",
            PdfConverter = PdfConverter.STANDARD,
            Model = "generic",
            Dictionaries = new List<string>
            {
                // provide list of dictionaries
            },
            // provide full path to the input and output folders
            InputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input"),
            OutputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output")
        };

        lwClient.TranslateBatchFile(translateBatchFileRequest);
    }

    private static void TranslateBatchFileWithLinguisticOptions(ILanguageWeaverClient lwClient)
    {
        Dictionary<string, string> linguisticOptions = new Dictionary<string, string>();
        linguisticOptions.Add("Spelling", "US");

        var translateBatchFileRequest = new TranslateBatchFileRequest()
        {
            SourceLanguageId = "fra",
            TargetLanguageId = "eng",
            Model = "generic",
            LinguisticOptions = linguisticOptions,
            Dictionaries = new List<string>
            {
                // provide list of dictionaries
            },
            // provide full path to the input and output folders
            InputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Input"),
            OutputBatchFolderPath = Path.Combine(
                Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName,
                "Resources", "Output")
        };

        lwClient.TranslateBatchFile(translateBatchFileRequest);
    }
}